java/query/src/main/java/sleeper/query/lambda/SqsQueryProcessor.java

```diff
@@ -120,12 +119,20 @@ public void processQuery(QueryOrLeafPartitionQuery query) {
     }
 
     private CloseableIterator<Record> processRangeQuery(Query query, TableProperties tableProperties, QueryStatusReportListeners queryTrackers) throws StateStoreException, QueryException {
+        // If the cache needs refreshing remove to allow for a new in initialisation
+        LOGGER.debug("Cache for table {}: {}", query.getTableName(), queryExecutorCache);
+        if (!queryExecutorCache.isEmpty() && queryExecutorCache.get(query.getTableName()).cacheRefreshRequired()) {
```

Comment:
This will throw a `NullPointerException` if a new table is created and queried before the cache is refreshed. You'll need to check if the table exists in the cache before checking if a cache refresh is required for the table.

Reply:
I think that scenario could be tested in SqsQueryProcessorLambdaIT by querying two separate tables.